### PR TITLE
Adding developer guide

### PR DESCRIPTION
**Pull request by martin-gaievski**
### Description
Adding developer guide for the repo, this should help onboard new contributors faster.

### Issues Resolved
https://github.com/opensearch-project/neural-search/issues/439

### Check List
- [X] All tests pass
- [X] Commits are signed as per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by martin-gaievski** [comment]
@ArnoldTingSu I know you were setting up the plugin recently, feel free to check and add what's missing

**Review by navneet1v — APPROVED** [review]

**Review by vibrantvarun — APPROVED** [review]

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    "lucene_version" : "9.8.0",
+    "minimum_wire_compatibility_version" : "2.12.0",
+    "minimum_index_compatibility_version" : "2.0.0"
+  },
+  "tagline" : "The OpenSearch Project: https://opensearch.org/"
+}
+```
+### Run Multi-node Cluster Locally
+
+It can be useful to test and debug on a multi-node cluster. In order to launch a 3 node cluster with the KNN plugin installed, run the following command:
```
[the remark this answers is not in the document]
Comment (martin-gaievski) [reply DEVELOPER_GUIDE.md:134]: ack

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+            },
+            "passage_text": {
+                "type": "text"
+            }
+        }
+    }
+}
+```
+
+The reason for this is that neural-search uses k-NN zip artifact from maven, and that artifact doesn't have native libraries. It's possible to install k-NN locally and build those native libraries (like Nmslib). For intruction on how to do this please check instruction from [k-NN plugin repository.](https://github.com/opensearch-project/k-NN)
```
Comment (ryanbogan) [inline DEVELOPER_GUIDE.md:216]: Small typo: instruction

Comment (martin-gaievski) [reply DEVELOPER_GUIDE.md:216]: ack

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+The test runner JVM will connect to a debugger attached to `localhost:8000` before running the tests.
+
+Additionally, it is possible to attach one debugger to the cluster JVM and another debugger to the test runner. First, make sure one debugger is listening on port `5005` and the other is listening on port `8000`. Then, run:
+```
+./gradlew :integTest -Dtest.debug=1 -Dcluster.debug=1
+```
+
+### Supported configurations
+
+Be default neural-search plugin supports `lucene` k-NN engine for local runs. Below is the sample request for creating of new index  using this engine:
```
[the remark this answers is not in the document]
Comment (martin-gaievski) [reply DEVELOPER_GUIDE.md]: ack

**Review by ryanbogan — APPROVED** [review]
LGTM!

**Pull request merged**